### ISSUE TITLE
Issue in compute_jackson_cheby_coeff

### ISSUE THREAD

**Issue by parkparkparkpark** [epfl-lts2/pygsp#74]
It seems to have to be corrected from 
`ch[0] = (2/(np.pi))*(np.arccos(filter_bounds[0])-np.arccos(filter_bounds[1]))`
to
`ch[0] = (1/(np.pi))*(np.arccos(filter_bounds[0])-np.arccos(filter_bounds[1]))`.

**Comment by nperraud** [comment]
Thanks for reporting this. 

@mdeff do you know why definition was used? I remember we had a big discussion about the definition when working on a student project together as it was not matching the definition of chebynet.

**Comment by mdeff** [comment]
Yes the [first coefficient is doubled in the PyGSP](https://github.com/epfl-lts2/pygsp/blob/70f47eaf5d4b521f3247c4f5f10e7df1f39b2b47/pygsp/filters/approximations.py#L103). In the ML code I wrote I did not double it indeed. The code in the PyGSP is a [straight translation from matlab](https://github.com/epfl-lts2/gspbox/blob/a7d9aac5e239f1bcb37a9bb09998cc161be2732f/utils/gsp_cheby_op.m#L105), which also doubles the first coefficient. Do you see any reason to double it @nperraud? Did you write the matlab code?

**Comment by mdeff** [comment]
I was going to not double the first coefficient in my [overhaul of the Chebyshev approximation](https://github.com/epfl-lts2/pygsp/blob/polynomial-approximations/pygsp/filters/approximations.py) in the `polynomial-approximations` branch.

**Comment by nperraud** [comment]
The Chebyshef implementation of the gspbox was taken from the code of the spectral wavelet paper from 2012. I do not know why this two is there and it might even be a mistake. I believe, it was intentionally done so, but I have no clue why.

**Comment by mdeff** [comment]
Thanks for the historical pointer. :) I don't think there's a mistake. It's a matter of definition. A factor of two must appear for the first coefficient. The coefficient is divided by 2 either when used (current PyGSP behavior) either when computed (the ML code, because the coefficient is learned not computed). I have a slight preference for the second. But is there a generally accepted definition?

**Comment by nperraud** [comment]
For reference here is the spectral graph wavelet toolbox: https://wiki.epfl.ch/sgwt 
And here is the spectral graph wavelet paper: https://www.sciencedirect.com/science/article/pii/S1063520310000552
The following shows the definition used for the Chebychev coefficient.
![image](https://user-images.githubusercontent.com/6399466/111783317-61bcf480-88ba-11eb-87bc-a5958a6c522d.png)
The pi/2 in eq. 52 and the 1/2 in eq. 53 should explain the convention used in the code.

Now I have not gone deeply to know why they chose this definition. I think we "simply" need to reference properly the definition we choose to use.

**Comment by mdeff** [comment]
>  The pi/2 in eq. 52 and the 1/2 in eq. 53 should explain the convention used in the code.

Agreed.

Now I have not gone deeply to know why they chose this definition. I think we "simply" need to reference properly the definition we choose to use.

Agreed. As [numpy](https://numpy.org/devdocs/reference/generated/numpy.polynomial.chebyshev.chebval.html) went without the factor 2 in the evaluation (i.e., like the ML code, unlike the current pygsp), I think we should follow this convention for consistency (and least surprise to users). Maybe even use numpy to compute the coefficients and evaluate the polynomial.

**Comment by nperraud** [comment]
Agreed, let us be compatible with numpy
